deps/verifier/src/tpm/mod.rs

```diff
@@ -0,0 +1,439 @@
+                if path.is_file() {
+                    // Only process files with .pub extension
+                    if let Some(extension) = path.extension() {
+                        if extension != "pub" {
+                            continue;
+                        }
+                    } else {
+                        // Skip files without extensions
+                        continue;
+                    }
+
+                    // Try to read as text first to avoid UTF-8 errors on binary files
+                    let key_content = match fs::read_to_string(&path) {
+                        Ok(content) => content,
+                        Err(_) => {
+                            // Skip files that can't be read as UTF-8 text
+                            debug!("Skipping non-UTF8 file: {:?}", path);
+                            continue;
+                        }
+                    };
+
+                    let pkey = PKey::public_key_from_pem(key_content.as_bytes())
+                        .context("Failed to parse PEM public key")?;
+                    let key_bytes = pkey.public_key_to_der()
+                        .context("Failed to convert PEM to DER")?;
+                    let hash = Sha256::digest(&key_bytes).to_vec();
+                    trusted_ak_hashes.insert(hash);
+                }
+            }
+        }
+        Ok(Self { trusted_ak_hashes })
+    }
+}
+
+fn verify_signature(quote: &Quote, ak_public: &str) -> Result<()> {
+    let pub_bytes = general_purpose::STANDARD
+        .decode(ak_public)
+        .context("Base64 decode of AK public failed")?;
+
+    let ak_pub = PKey::public_key_from_der(&pub_bytes).context("Failed to parse AK public key")?;
+
+    quote
+        .verify_signature(&ak_pub)
+        .context("TPM quote signature verification failed")?;
+
+    debug!("TPM quote signature is valid.");
+    Ok(())
+}
+
+fn verify_pcrs(quote: &Quote) -> Result<()> {
+    quote
+        .verify_pcrs()
+        .context("Digest of PCRs does not match digest in Quote")?;
+    debug!("PCR verification completed successfully");
+    Ok(())
+}
+
+fn verify_nonce(quote: &Quote, expected_report_data: &[u8]) -> Result<()> {
+    let nonce = quote.nonce()?;
+
+    debug!("Expected report_data ({} bytes): {}", expected_report_data.len(), hex::encode(expected_report_data));
+    debug!("Quote nonce ({} bytes): {}", nonce.len(), hex::encode(&nonce));
+
+    // TPM attester pads report_data to 64 bytes, so we need to pad expected_report_data the same way
+    const TPM_REPORT_DATA_SIZE: usize = 64;
+    let mut padded_expected = expected_report_data.to_vec();
+    padded_expected.resize(TPM_REPORT_DATA_SIZE, 0);
+
+
+    match nonce == padded_expected {
+        true => {
+            debug!("TPM report_data verification completed successfully");
+            Ok(())
+        }
+        false => {
+            debug!("Nonce and padded expected are different");
+            Err(TpmVerifierError::NonceMismatch.into())
```

Comment:
Looks like we are mixing `anyhow` and `thiserror` in the code. The `Err(TpmVerifierError::NonceMismatch.into())` here is equal to 
`bail!("TPM quote nonce doesn't match expected report_data")`.

Usually, if we want to use thiserror, we can define a `Result<T> = std::result::Result<T, TpmVerifierError>` and return the new `Result` in all `pub` functions. If we want to return the `anyhow::Result`, we do not need to define the consts but directly write error message in-flight.

I suggest to get rid of `thiserror`and use anyhow for all places to make it simpler